src/main/java/sirius/biz/tenants/TenantUserManager.java

```diff
@@ -764,7 +766,13 @@ public void onExternalLogin(WebContext ctx, UserInfo user) {
         recordLogin(user, true);
     }
 
-    protected abstract void recordLogin(UserInfo user, boolean external);
+    /**
+     * Actually tracks the login of the user in the account entity.
+     *
+     * @param user     the user which logged in
+     * @param external whether or not the login was from an external source (e.g. via SAML)
+     */
+    public abstract void recordLogin(UserInfo user, boolean external);
```

Comment:
Ich wäre fast dafür, das in `UserManager` hoch zu ziehen. Kann man ja im `PublicUserManager` dann auch als No-Op implementieren. Hätte in S2 den Charme, dass das auch für Frontend-Nutzer funktionieren würde (die bei uns in einem gewissen Umfang auf FTP zugreifen können)

Reply:
Macht natürlich den PR komplizierter (weil Web), aber man spart sich auch den blöden Cast oben